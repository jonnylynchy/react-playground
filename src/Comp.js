import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Comp extends Component {
	static propTypes = {
		render: PropTypes.func.isRequired,
	};

	state = {
		message: "You haven't clicked yet..."
	};

	handleClick = _ => {
		this.setState({
			message: "Clickity click click yo!"
		});
	};

	render() {
		return (
			<div style={{ height: '100%' }} onClick={this.handleClick}>
				{this.props.render(this.state)}
			</div>
		);
	}
}

export default Comp;