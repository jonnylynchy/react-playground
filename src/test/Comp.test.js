import React from 'react';
import { shallow } from 'enzyme';

import Comp from '../Comp';

const TestComp = () => <div>Test Component</div>;

describe('when <Comp /> renders', () => {

	it('should render some markup and a component', () => {
		const renderFunc = state =>
			<React.Fragment>
				<h1>{state.message}</h1>
				<TestComp />
			</React.Fragment>;

		const wrapper = shallow(<Comp render={renderFunc} />);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should render something different on click', () => {
		const renderFunc = state =>
			<React.Fragment>
				<h1>{state.message}</h1>
			</React.Fragment>;

		const wrapper = shallow(<Comp render={renderFunc} />);
		wrapper.find('div').simulate('click');

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should match json serialized snapshot', () => {
		const renderFunc = state =>
			<React.Fragment>
				<h1>{state.message}</h1>
				<TestComp />
			</React.Fragment>;

		const wrapper = shallow(<Comp render={renderFunc} />);

		expect(wrapper).toMatchSnapshot();
	});

});