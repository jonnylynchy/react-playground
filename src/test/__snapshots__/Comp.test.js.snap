// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when <Comp /> renders should match json serialized snapshot 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "height": "100%",
    }
  }
>
  <h1>
    You haven't clicked yet...
  </h1>
  <TestComp />
</div>
`;

exports[`when <Comp /> renders should render some markup and a component 1`] = `
"<div style={{...}} onClick={[Function]}>
  <h1>
    You haven&#39;t clicked yet...
  </h1>
  <TestComp />
</div>"
`;

exports[`when <Comp /> renders should render something different on click 1`] = `
"<div style={{...}} onClick={[Function]}>
  <h1>
    Clickity click click yo!
  </h1>
</div>"
`;
